engine/classes/ElggEntity.php

```diff
@@ -1622,38 +1620,33 @@ public function disable($reason = "", $recursive = true) {
 		$guid = (int) $this->guid;
 
 		if ($recursive) {
-			// Only disable enabled subentities
-			$hidden = access_get_show_hidden_status();
-			access_show_hidden_entities(false);
-
-			$ia = elgg_set_ignore_access(true);
-
-			$base_options = [
-				'wheres' => [
-					"e.guid != $guid",
-				],
-				'limit' => false,
-			];
-
-			foreach (['owner_guid', 'container_guid'] as $db_column) {
-				$options = $base_options;
-				$options[$db_column] = $guid;
-
-				$subentities = new \ElggBatch('elgg_get_entities', $options);
-				$subentities->setIncrementOffset(false);
-
-				foreach ($subentities as $subentity) {
-					/* @var $subentity \ElggEntity */
-					if (!$subentity->isEnabled()) {
-						continue;
+			$disable_relatives = function () use ($guid, $reason) {
```

Comment:
I think when we define the callable like this, we should define $flags directly above it, to help readers understand the environment in which it'll run.